spec/jobs/order_cycle_notification_job_spec.rb

```diff
@@ -4,20 +4,18 @@
 
 RSpec.describe OrderCycleNotificationJob do
   let(:order_cycle) { create(:order_cycle) }
-  let(:mail) { double(:mail, deliver_now: true) }
-
-  before do
-    allow(ProducerMailer).to receive(:order_cycle_report).twice.and_return(mail)
-  end
 
   it "sends a mail to each supplier" do
-    OrderCycleNotificationJob.perform_now order_cycle.id
-    expect(ProducerMailer).to have_received(:order_cycle_report).twice
+    expect {
+      OrderCycleNotificationJob.perform_now(order_cycle.id)
+    }.to enqueue_mail(ProducerMailer, :order_cycle_report).twice
   end
 
   it "records that mails have been sent for the order cycle" do
-    expect do
+    expect {
       OrderCycleNotificationJob.perform_now(order_cycle.id)
-    end.to change{ order_cycle.reload.mails_sent? }.from(false).to(true)
+    }.to change {
+      order_cycle.reload.mails_sent?
```

Comment:
Hmm, now `mails_sent` means "mails enqueued".

Before, it was run in serial so we could know at the end if mails were all sent successfully or not. Now we don't know. Does that matter?

It looks like `mails_sent` is used to d
1. display an email success icon on the order cycle edit page, and 
2. enable a "re-notify" button.

I think no. 1 would be best served by a "mails sent" , and no2 would be better as "mails enqueued". So.. I think it's all close enough that it doesn't matter. 👍